Bottom: 0a9c637fd7176853cb898de9f4b782a21a02d500
Top:    5445f3d8e7e02b1d02e5f1c8eff723d31b77c9e4
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-26 00:16:01 +0100

Refresh of draw-c-split-gathering-routine

---

diff --git a/src/draw.c b/src/draw.c
index 3d85e07..a638bd7 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -1522,23 +1522,34 @@ DrawElementPackage (ElementTypePtr Element)
 void
 DrawElementPinsAndPads (ElementTypePtr Element)
 {
-  PAD_LOOP (Element);
-  {
-    if (doing_pinout || doing_assy || FRONT (pad) || PCB->InvisibleObjectsOn)
-      if (Gathering)
-        DrawPad (pad);
-      else
-        draw_pad (pad);
-  }
-  END_LOOP;
-  PIN_LOOP (Element);
-  {
-    if (Gathering)
-      DrawPin (pin);
-    else
-      draw_pin (pin, true);
-  }
-  END_LOOP;
+  if (Gathering)
+    {
+      PAD_LOOP (Element);
+      {
+        if (doing_pinout || doing_assy || FRONT (pad) || PCB->InvisibleObjectsOn)
+          DrawPad (pad);
+      }
+      END_LOOP;
+      PIN_LOOP (Element);
+      {
+        DrawPin (pin);
+      }
+      END_LOOP;
+    }
+  else
+    {
+      PAD_LOOP (Element);
+      {
+        if (doing_pinout || doing_assy || FRONT (pad) || PCB->InvisibleObjectsOn)
+          draw_pad (pad);
+      }
+      END_LOOP;
+      PIN_LOOP (Element);
+      {
+        draw_pin (pin, true);
+      }
+      END_LOOP;
+    }
 }
 
 /* ---------------------------------------------------------------------------
